### PR TITLE
docs: updates README for Pluggable Auth 

### PR DESCRIPTION
**Pull request by lsirac**
Adds documentation for Pluggable Auth which is being added in https://github.com/googleapis/google-auth-library-java/pull/908.

### CODE REVIEW

**Review by Neenu1995 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -323,6 +323,116 @@ request to `$URL_TO_GET_OIDC_TOKEN`, e.g. `Metadata-Flavor=Google`.
 You can now [use the Auth library](#using-external-identities) to call Google Cloud
 resources from an OIDC provider.
 
+#### Using Executable-sourced credentials with OIDC and SAML
```
Comment (TimurSadykov) [inline README.md:326]: Please add a link to it to table on contents line 24

File: README.md [hunk 2]
```diff
@@ -323,6 +323,116 @@ request to `$URL_TO_GET_OIDC_TOKEN`, e.g. `Metadata-Flavor=Google`.
...
+    When false, the response must contain the error code and message fields and exit with a non-zero value.
+  * `token_type`: The 3rd party subject token type. Must be *urn:ietf:params:oauth:token-type:jwt*, 
+     *urn:ietf:params:oauth:token-type:id_token*, or *urn:ietf:params:oauth:token-type:saml2*.
+  * `id_token`: The 3rd party OIDC token.
+  * `saml_response`: The 3rd party SAML response.
+  * `expiration_time`: The 3rd party subject token expiration time in seconds (unix epoch time).
+  * `code`: The error code string.
+  * `message`: The error message.
+
+All response types must include both the `version` and `success` fields.
```
Comment (TimurSadykov) [inline README.md:412]: **the** success

File: README.md [hunk 3]
```diff
@@ -323,6 +323,116 @@ request to `$URL_TO_GET_OIDC_TOKEN`, e.g. `Metadata-Flavor=Google`.
...
+ * Successful responses must include the `token_type`, `expiration_time`, and one of
+   `id_token` or `saml_response`.
+ * Error responses must include both the `code` and `message` fields.
```
Comment (TimurSadykov) [inline README.md:415]: **the** message

**Review by TimurSadykov — APPROVED** [review]
LGTM with some nits

**Pull request merged**